infra/src/main/resources/db/migration/prod/V96__add_design_cancelled_flag.sql

```diff
@@ -0,0 +1,106 @@
+alter table layout.track_number
+  disable trigger version_update_trigger;
+alter table layout.track_number
+  disable trigger version_row_trigger;
+
+alter table layout.reference_line
+  disable trigger version_update_trigger;
+alter table layout.reference_line
+  disable trigger version_row_trigger;
+
+alter table layout.location_track
+  disable trigger version_update_trigger;
+alter table layout.location_track
+  disable trigger version_row_trigger;
+
+alter table layout.switch
+  disable trigger version_update_trigger;
+alter table layout.switch
+  disable trigger version_row_trigger;
+
+alter table layout.km_post
+  disable trigger version_update_trigger;
+alter table layout.km_post
+  disable trigger version_row_trigger;
+
+alter table layout.track_number
+  add column cancelled boolean not null default false;
+
+alter table layout.track_number_version
+  add column cancelled boolean null;
+update layout.track_number_version set cancelled = false where true;
```

Comment:
Tuo versiotaulun `not null` pisti silmään. Onko sille jokin tarkoitus, kun kaikki olemassaolevat rivit käydään kuitenkin päivittämässä heti `false`:iksi? 🤔 Hieman jännää että versio- ja ei-versiotaulujen välillä on tällainen eroavaisuus.

Reply:
Molemmathan tulee siis not nulliksi, versiotaulussa sillä kylläkin on vähemmän merkitystä. Varsinainen ero on siinä, onko defaulttiarvoa: Live-taululla on, versiotaululla ei. Minä tapaan mielelläni määrittää default-arvoja live-tauluissa, koska tällöin niitä on helpompi käsitellä erinäisissä testaus- ja korjaustilanteissa käsin, mutta versiotaulu ei niitä tarvitse. Nulliusero tulee sitten vaan siitä, että jos defaulttiarvoa ei aseta, ei pysty asettamaan not nulliakaan ennen kuin jokaisella rivillä on arvo.